src/exe/cimbar_recv/recv.cpp

```diff
@@ -130,29 +136,41 @@ int main(int argc, char** argv)
 			continue;
 		}
 
-		cv::UMat img = mat.getUMat(cv::ACCESS_RW).clone();
+		cv::Mat img = mat.clone();
 		cv::cvtColor(mat, mat, cv::COLOR_BGR2RGB);
 
 		// draw some stats on mat?
 		window.show(mat, 0);
 
-		// extract
-		bool shouldPreprocess = false;
-		int res = ext.extract(img, img);
-		if (!res)
+		// extract, decode, etc
+		int bytes = cimbard_scan_extract_decode(img.data, img.cols, img.rows, 3, bufspace.data(), bufspace.size());
+		if (bytes <= 0)
+			continue;
+
+		if (bytes % chunkSize != 0)
 		{
-			//std::cerr << "no extract " << mat.cols << "," << mat.rows << std::endl;
+			std::cerr << "WEIRD SEEMS BAD? " << count << std::endl;
 			continue;
 		}
-		else if (res == Extractor::NEEDS_SHARPEN)
-			shouldPreprocess = true;
 
-		// decode
-		int bytes = dec.decode_fountain(img, sink, color_mode, shouldPreprocess);
-		if (bytes > 0)
-			std::cerr << "got some bytes " << bytes << std::endl;
+		int64_t res = cimbard_fountain_decode(bufspace.data(), bytes);
+		if (res > 0)
+		{
+			// attempt save
+			uint32_t fileId = res;
 
-		std::cerr << turbo::str::join(sink.get_progress()) << std::endl;
+			unsigned size = cimbard_get_filesize(fileId);
+			std::string filename = fmt::format("0.{}", size);
```

Comment:
We should *probably* should pull the encode_id out of the fileId for this filename. :thinking: 

(actual filename support is coming)